
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title> - bio4j</title>
  <meta name="author" content="oh no sequences!">

  
  <meta name="description" content="                  Bio4j accepted as a mentoring org for Google Summer of Code 2014!! check the blog postBio4j is a high-performance cloud-enabled g...">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://bio4j.com/">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="/javascripts/ender.js"></script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <link href="/atom.xml" rel="alternate" title="bio4j" type="application/atom+xml">
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href='http://fonts.googleapis.com/css?family=Open+Sans:600italic,300italic,400,400italic,600,300' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bree+Serif' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bitter:400,400italic' rel='stylesheet' type='text/css'>
<!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  showMathMenu: false,
  TeX: {
    extensions: ["AMSmath.js", "AMSsymbols.js"]
  }, 
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", webFont: "TeX", availableFonts: [] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>


  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-12161444-3']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   >
  <header role="banner"><!-- <hgroup>
  <h1><a href="/">bio4j</a></h1>
  
    <h2>bio data graph db</h2>
  
</hgroup>
 -->

 <hgroup>
  
    <a href="/">
      <img src="/images/logo.png">
    </a>
  <!-- <h2>bio data graph db</h2> -->

</hgroup>
</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
<form action="http://google.com/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:bio4j.com" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
<ul class="main-navigation">
  <li><a href="/blog">blog</a></li>
  <li><a href="/blog/archives">archives</a></li>
  <li><a href="/code">code</a></li>
  <li><a href="/docs">docs</a></li>
  <li><a href="/community">community</a></li>
  <li><a href="/about">about</a></li>
  </ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article role="article">
  
  <header>
    <h1 class="entry-title"></h1>
    
  </header>
  
  <p><em>Bio4j accepted as a mentoring org for Google Summer of Code 2014!! check the <a href="/blog/2014/02/bio4j-accepted-for-google-summer-of-code-2014">blog post</a></em></p>

<p>Bio4j is a high-<strong>performance</strong> <strong>cloud</strong>-enabled <strong>graph</strong>-based <strong>open</strong> source <strong>bioinformatics data</strong> platform, integrating the data available in the most representative open data sources around protein information. It integrates the data available in </p>
<ul>
<li><a href="http://www.uniprot.org/help/uniprotkb">UniProtKB</a> (<a href="http://web.expasy.org/docs/swiss-prot_guideline.html">SwissProt</a> + <a href="https://www.ebi.ac.uk/uniprot/TrEMBLstats">Trembl</a>)</li>
<li><a href="http://www.geneontology.org/">Gene Ontology</a> (GO)</li>
<li><a href="http://www.uniprot.org/help/uniref">UniRef</a> (50, 90, 100)</li>
<li><a href="http://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a></li>
<li><a href="http://www.ncbi.nlm.nih.gov/taxonomy">NCBI taxonomy</a></li>
<li><a href="http://enzyme.expasy.org/">Expasy Enzyme</a></li>
</ul>
<p>Bio4j provides a completely new and powerful <strong>framework for protein related information querying and management</strong>. 
Since it relies on a high-performance graph engine, data is stored in a way that semantically represents its own structure. 
On the contrary, traditional relational databases must flatten the data they represent into tables, creating <em>artificial</em> ids in order to connect the different tuples; which can in some cases eventually lead to domain models that have almost nothing to do with the actual structure of data.</p>
<p>This set of slides from <a href="https://fosdem.org/2014/schedule/event/graphdevroom_bio4j_2/">FOSDEM 2014</a> should give an up-to-date overview of the project:</p>
<ul>
<li><strong><a href="http://ohnosequences.com/slides/fosdem-2014/bio4j-bigger-faster-leaner">Bio4j: bigger, faster, leaner - FOSDEM GraphDevRoom 2014</a></strong></li>
</ul>
<h3>scalability</h3>
<ul>
<li>
<p>First of all, Bio4j has an <a href="https://github.com/bio4j/bio4j"><strong>Abstract Domain Model</strong></a>, which allows you to use it without binding to a particular backend implementation.</p>
</li>
<li>
<p>Next, it has an intermediate <a href="https://github.com/bio4j/blueprints"><strong>Blueprints layer</strong></a>, which allows us to make a default implementation of the abstract interface using <a href="https://github.com/tinkerpop/blueprints/wiki">Tinkerpop Blueprints API</a> and at the same time stay independent from the choice of database technology.</p>
</li>
<li>
<p>And finally, there are technology specific versions:</p>
</li>
<li><a href="https://github.com/bio4j/titandb"><strong>Titan DB implementation</strong></a></li>
<li><a href="https://github.com/bio4j/neo4jdb"><strong>Neo4j DB implementation</strong></a></li>
</ul>
<h3>modularity</h3>
<p>Bio4j includes a few different data sources and you may not always be interested in having all of them together. Thatâ€™s why the importing process is modular and customizable, allowing you to import just the data you are interested in.</p>
<p>Also, Bio4j has <a href="https://github.com/bio4j/modules">Statika-based module system</a>, which dramatically simplifies the process of building and deploying custom releases of Bio4j.</p>
<h3>performance</h3>
<p>In Bio4j data is organized in a way semantically equivalent to what it represents thanks to the graph structure. That means that queries which would even be impossible to perform with a standard Relational DB, can be feasible with Bio4j obtaining good performance results.</p>
<h3>licensing</h3>
<p>Bio4j is an <strong>open source</strong> platform released under <a href="http://www.gnu.org/licenses/agpl.html"><strong>AGPLv3</strong></a>.</p>

  
    <footer>
      
      
        <div class="sharing">
  
  
  
</div>

      
    </footer>
  
</article>

</div>

<aside class="sidebar">
  
    
<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating...</li>
  </ul>
  
  <a href="https://github.com/bio4j">@bio4j</a> on GitHub
  
  <script type="text/javascript">
    $.domReady(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'bio4j',
            count: 0,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>



  
</aside>


    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2015 - oh no sequences! -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
</p>

</footer>
  


<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>




  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = 'http://platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
