<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_19) on Wed Oct 24 11:41:08 CEST 2012 -->
<META http-equiv="Content-Type" content="text/html; charset=UTF-8">
<TITLE>
com.era7.bioinfo.bio4j.model.relationships.features (Bio4j 0.8 API)
</TITLE>

<META NAME="date" CONTENT="2012-10-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="com.era7.bioinfo.bio4j.model.relationships.features (Bio4j 0.8 API)";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Package</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-use.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/comment/package-summary.html"><B>PREV PACKAGE</B></A>&nbsp;
&nbsp;<A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/go/package-summary.html"><B>NEXT PACKAGE</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../../index.html?com/era7/bioinfo/bio4j/model/relationships/features/package-summary.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="package-summary.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<H2>
Package com.era7.bioinfo.bio4j.model.relationships.features
</H2>
Protein features relationships
<P>
<B>See:</B>
<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<A HREF="#package_description"><B>Description</B></A>
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Class Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/ActiveSiteFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">ActiveSiteFeatureRel</A></B></TD>
<TD>Amino acid(s) directly involved in the activity of an enzyme</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/BasicFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">BasicFeatureRel</A></B></TD>
<TD>Basic feature relationship</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/BindingSiteFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">BindingSiteFeatureRel</A></B></TD>
<TD>Binding site for any chemical group (co-enzyme, prosthetic group, etc.)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/CalciumBindingRegionFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">CalciumBindingRegionFeatureRel</A></B></TD>
<TD>Position(s) of calcium binding region(s) within the protein</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/ChainFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">ChainFeatureRel</A></B></TD>
<TD>Extent of a polypeptide chain in the mature protein</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/CoiledCoilRegionFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">CoiledCoilRegionFeatureRel</A></B></TD>
<TD>Positions of regions of coiled coil within the protein</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/CompositionallyBiasedRegionFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">CompositionallyBiasedRegionFeatureRel</A></B></TD>
<TD>Region of compositional bias in the protein</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/CrossLinkFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">CrossLinkFeatureRel</A></B></TD>
<TD>Residues participating in covalent linkage(s) between proteins</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/DisulfideBondFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">DisulfideBondFeatureRel</A></B></TD>
<TD>Cysteine residues participating in disulfide bonds</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/DnaBindingRegionFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">DnaBindingRegionFeatureRel</A></B></TD>
<TD>Position and type of a DNA-binding domain</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/DomainFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">DomainFeatureRel</A></B></TD>
<TD>Position and type of each modular protein domain</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/GlycosylationSiteFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">GlycosylationSiteFeatureRel</A></B></TD>
<TD>Covalently attached glycan group(s)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/HelixFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">HelixFeatureRel</A></B></TD>
<TD>Helical regions within the experimentally determined protein structure</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/InitiatorMethionineFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">InitiatorMethionineFeatureRel</A></B></TD>
<TD>Cleavage of the initiator methionine</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/IntramembraneRegionFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">IntramembraneRegionFeatureRel</A></B></TD>
<TD>Extent of a region located in a membrane without crossing it</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/LipidMoietyBindingRegionFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">LipidMoietyBindingRegionFeatureRel</A></B></TD>
<TD>Covalently attached lipid group(s)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/MetalIonBindingSiteFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">MetalIonBindingSiteFeatureRel</A></B></TD>
<TD>Binding site for a metal ion</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/ModifiedResidueFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">ModifiedResidueFeatureRel</A></B></TD>
<TD>Modified residues excluding lipids, glycans and protein cross-links</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/MutagenesisSiteFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">MutagenesisSiteFeatureRel</A></B></TD>
<TD>Site which has been experimentally altered by mutagenesis</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/NonConsecutiveResiduesFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">NonConsecutiveResiduesFeatureRel</A></B></TD>
<TD>Indicates that two residues in a sequence are not consecutive</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/NonStandardAminoAcidFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">NonStandardAminoAcidFeatureRel</A></B></TD>
<TD>Occurence of non-standard amino acids (selenocysteine and pyrrolysine) in the protein sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/NonTerminalResidueFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">NonTerminalResidueFeatureRel</A></B></TD>
<TD>The sequence is incomplete.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/NucleotidePhosphateBindingRegionFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">NucleotidePhosphateBindingRegionFeatureRel</A></B></TD>
<TD>Nucleotide phosphate binding region</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/PeptideFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">PeptideFeatureRel</A></B></TD>
<TD>Extent of an active peptide in the mature protein</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/PropeptideFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">PropeptideFeatureRel</A></B></TD>
<TD>Part of a protein that is cleaved during maturation or activation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/RegionOfInterestFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">RegionOfInterestFeatureRel</A></B></TD>
<TD>Region of interest in the sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/RepeatFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">RepeatFeatureRel</A></B></TD>
<TD>Positions of repeated sequence motifs or repeated domains</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/SequenceConflictFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">SequenceConflictFeatureRel</A></B></TD>
<TD>Description of sequence discrepancies of unknown origin</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/SequenceVariantFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">SequenceVariantFeatureRel</A></B></TD>
<TD>Description of a natural variant of the protein</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/ShortSequenceMotifFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">ShortSequenceMotifFeatureRel</A></B></TD>
<TD>Short (up to 20 amino acids) sequence motif of biological interest</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/SignalPeptideFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">SignalPeptideFeatureRel</A></B></TD>
<TD>Sequence targeting proteins to the secretory pathway or periplasmic space</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/SiteFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">SiteFeatureRel</A></B></TD>
<TD>Any interesting single amino acid site on the sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/SpliceVariantFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">SpliceVariantFeatureRel</A></B></TD>
<TD>Amino acid change(s) producing alternate protein isoforms</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/StrandFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">StrandFeatureRel</A></B></TD>
<TD>Beta strand regions within the experimentally determined protein structure</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/TopologicalDomainFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">TopologicalDomainFeatureRel</A></B></TD>
<TD>Location of non-membrane regions of membrane-spanning proteins</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/TransitPeptideFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">TransitPeptideFeatureRel</A></B></TD>
<TD>Extent of a transit peptide for organelle targeting</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/TransmembraneRegionFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">TransmembraneRegionFeatureRel</A></B></TD>
<TD>Extent of a membrane-spanning region</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/TurnFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">TurnFeatureRel</A></B></TD>
<TD>Turns within the experimentally determined protein structure</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/UnsureResidueFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">UnsureResidueFeatureRel</A></B></TD>
<TD>Regions of uncertainty in the sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD WIDTH="15%"><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/ZincFingerRegionFeatureRel.html" title="class in com.era7.bioinfo.bio4j.model.relationships.features">ZincFingerRegionFeatureRel</A></B></TD>
<TD>Position(s) and type(s) of zinc fingers within the protein</TD>
</TR>
</TABLE>
&nbsp;

<P>
<A NAME="package_description"><!-- --></A><H2>
Package com.era7.bioinfo.bio4j.model.relationships.features Description
</H2>

<P>
Protein features relationships
<P>

<P>
<DL>
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Package</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-use.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/comment/package-summary.html"><B>PREV PACKAGE</B></A>&nbsp;
&nbsp;<A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/go/package-summary.html"><B>NEXT PACKAGE</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../../index.html?com/era7/bioinfo/bio4j/model/relationships/features/package-summary.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="package-summary.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>
Copyright &#169; 2012. All Rights Reserved.
</BODY>
</HTML>
