<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_19) on Wed Oct 24 11:41:04 CEST 2012 -->
<META http-equiv="Content-Type" content="text/html; charset=UTF-8">
<TITLE>
FeatureTypeNode (Bio4j 0.8 API)
</TITLE>

<META NAME="date" CONTENT="2012-10-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="FeatureTypeNode (Bio4j 0.8 API)";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/FeatureTypeNode.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/EnzymeNode.html" title="class in com.era7.bioinfo.bio4j.model.nodes"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/GoTermNode.html" title="class in com.era7.bioinfo.bio4j.model.nodes"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../index.html?com/era7/bioinfo/bio4j/model/nodes/FeatureTypeNode.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="FeatureTypeNode.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
com.era7.bioinfo.bio4j.model.nodes</FONT>
<BR>
Class FeatureTypeNode</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../../../resources/inherit.gif" ALT="extended by ">com.era7.bioinfo.bioinfoneo4j.BasicEntity
      <IMG SRC="../../../../../../resources/inherit.gif" ALT="extended by "><B>com.era7.bioinfo.bio4j.model.nodes.FeatureTypeNode</B>
</PRE>
<HR>
<DL>
<DT><PRE>public class <B>FeatureTypeNode</B><DT>extends com.era7.bioinfo.bioinfoneo4j.BasicEntity</DL>
</PRE>

<P>
This class models precise but simple means for the annotation of sequence data.
 It describes regions or sites of interest in the protein sequence.
 In general it lists post-translational modifications, binding sites, enzyme active sites,
 local secondary structure or other characteristics reported in the cited references.
 Sequence conflicts between references are also included in this section. <br>
 Features modelled list:
 <br><br><b>Molecule processing</b><br>
 <table>
 <tr><td><b>Initiator methionine:</b></td><td>Cleavage of the initiator methionine</td></tr>
 <tr><td><b>Signal:</b></td><td>Sequence targeting proteins to the secretory pathway or periplasmic space</td></tr>
 <tr><td><b>Transit peptide:</b></td><td>Extent of a transit peptide for organelle targeting</td></tr>
 <tr><td><b>Propeptide:</b></td><td>Part of a protein that is cleaved during maturation or activation </td></tr>
 <tr><td><b>Chain:</b></td><td>Extent of a polypeptide chain in the mature protein</td></tr>
 <tr><td><b>Peptide:</b></td><td>Extent of an active peptide in the mature protein</td></tr>
 </table>
 <br><b>Regions</b><br><br>
 <table>
 <tr><td><b>Topological domain:</b></td><td> Location of non-membrane regions of membrane-spanning proteins</td></tr>
 <tr><td><b>Transmembrane:</b></td><td> Extent of a membrane-spanning region</td></tr>
 <tr><td><b>Intramembrane:</b></td><td> Extent of a region located in a membrane without crossing it</td></tr>
 <tr><td><b>Domain:</b></td><td> Position and type of each modular protein domain</td></tr>
 <tr><td><b>Repeat:</b></td><td> Positions of repeated sequence motifs or repeated domains</td></tr>
 <tr><td><b>Calcium binding:</b></td><td> Position(s) of calcium binding region(s) within the protein</td></tr>
 <tr><td><b>Zinc finger:</b></td><td> Position(s) and type(s) of zinc fingers within the protein</td></tr>
 <tr><td><b>DNA binding:</b></td><td> Position and type of a DNA-binding domain</td></tr>
 <tr><td><b>Nucleotide binding:</b></td><td> Nucleotide phosphate binding region</td></tr>
 <tr><td><b>Region:</b></td><td> Region of interest in the sequence</td></tr>
 <tr><td><b>Coiled coil:</b></td><td> Positions of regions of coiled coil whithin the protein</td></tr>
 <tr><td><b>Motif:</b></td><td> Short (up to 20 amino acids) sequence motif of biological interest</td></tr>
 <tr><td><b>Compositional bias:</b></td><td> Region of compositional bias in the protein</td></tr>
 </table>
 <br><b>Sites</b><br><br>
 <table>
 <tr><td><b>Active site:</b></td><td> Amino acid(s) directly involved in the activity of an enzyme</td></tr>
 <tr><td><b>Metal binding:</b></td><td> Binding site for a metal ion</td></tr>
 <tr><td><b>Binding site:</b></td><td> Binding site for any chemical group (co-enzyme, prosthetic group, ect.)</td></tr>
 <tr><td><b>Site:</b></td><td> Any interesting single amino acid site on the sequence</td></tr>
 </table>
 <br><b>Amino acid modifications</b><br><br>
 <table>
 <tr><td><b>Non-standard residue:</b></td><td> Occurrence of non-standard amino acids (selenocysteine an pyrrolysine) in the protein sequence</td></tr>
 <tr><td><b>Modified residue:</b></td><td> Modified residues excluding lipids, glycans and protein cross-links</td></tr>
 <tr><td><b>Lipidation:</b></td><td> Covalently attached lipid group(s)</td></tr>
 <tr><td><b>Glycosylation:</b></td><td> Covalently attached glycan group(s)</td></tr>
 <tr><td><b>Disulfide bond:</b></td><td> Cysteine residues participating in disulfide bonds</td></tr>
 <tr><td><b>Cross-link:</b></td><td> Residues participating in covalent linkage(s) between proteins</td></tr>
 </table>
 <br><b>Natural variations</b><br><br>
 <table>
 <tr><td><b>Alternative sequence:</b></td><td> Amino acid change(s) producing alternate protein isoforms</td></tr>
 <tr><td><b>Natural variant:</b></td><td> Description of a natural variant of the protein</td></tr>
 </table>
 <br><b>Experimental info</b><br><br>
 <table>
 <tr><td><b>Mutagenesis:</b></td><td> Site which has been experimentally altered by mutagenesis</td></tr>
 <tr><td><b>Sequence uncertainty:</b></td><td> Regions of uncertainty in the sequence</td></tr>
 <tr><td><b>Sequence conflict:</b></td><td> Description of sequence discrepancies of unknown origin</td></tr>
 <tr><td><b>Non-adjacent residues:</b></td><td> Indicates that two residues in a sequence are not consecutive</td></tr>
 <tr><td><b>Non-terminal residue:</b></td><td> The sequence is incomplete. Indicate that a residue is not the terminal residue of the complete protein</td></tr>
 </table>
 <br><b>Secondary structure</b><br><br>
 <table>
 <tr><td><b>Helix:</b></td><td> Helical regions within the experimentally determined protein structure</td></tr>
 <tr><td><b>Turn:</b></td><td> Turns within the experimentally determined protein structure</td></tr>
 <tr><td><b>Beta strand:</b></td><td> Beta strand regions within the experimentally determined protein structure</td></tr>
 </table>
<P>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>Pablo Pareja Tobes <ppareja@era7.com></DD>
</DL>
<HR>

<P>
<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/FeatureTypeNode.html#FEATURE_TYPE_NAME_INDEX">FEATURE_TYPE_NAME_INDEX</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/FeatureTypeNode.html#NAME_PROPERTY">NAME_PROPERTY</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Feature type name</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/FeatureTypeNode.html#NODE_TYPE">NODE_TYPE</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="fields_inherited_from_class_com.era7.bioinfo.bioinfoneo4j.BasicEntity"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Fields inherited from class com.era7.bioinfo.bioinfoneo4j.BasicEntity</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>node, NODE_TYPE_PROPERTY</CODE></TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/FeatureTypeNode.html#FeatureTypeNode(org.neo4j.graphdb.Node)">FeatureTypeNode</A></B>(org.neo4j.graphdb.Node&nbsp;n)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/FeatureTypeNode.html#equals(java.lang.Object)">equals</A></B>(java.lang.Object&nbsp;obj)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/FeatureTypeNode.html#getName()">getName</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/FeatureTypeNode.html#hashCode()">hashCode</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/FeatureTypeNode.html#setName(java.lang.String)">setName</A></B>(java.lang.String&nbsp;value)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/FeatureTypeNode.html#toString()">toString</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_com.era7.bioinfo.bioinfoneo4j.BasicEntity"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class com.era7.bioinfo.bioinfoneo4j.BasicEntity</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>getNode, getNodeType, setNodeType</CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, finalize, getClass, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="FEATURE_TYPE_NAME_INDEX"><!-- --></A><H3>
FEATURE_TYPE_NAME_INDEX</H3>
<PRE>
public static final java.lang.String <B>FEATURE_TYPE_NAME_INDEX</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../../../../constant-values.html#com.era7.bioinfo.bio4j.model.nodes.FeatureTypeNode.FEATURE_TYPE_NAME_INDEX">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="NODE_TYPE"><!-- --></A><H3>
NODE_TYPE</H3>
<PRE>
public static final java.lang.String <B>NODE_TYPE</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="NAME_PROPERTY"><!-- --></A><H3>
NAME_PROPERTY</H3>
<PRE>
public static final java.lang.String <B>NAME_PROPERTY</B></PRE>
<DL>
<DD>Feature type name
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../../../../constant-values.html#com.era7.bioinfo.bio4j.model.nodes.FeatureTypeNode.NAME_PROPERTY">Constant Field Values</A></DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="FeatureTypeNode(org.neo4j.graphdb.Node)"><!-- --></A><H3>
FeatureTypeNode</H3>
<PRE>
public <B>FeatureTypeNode</B>(org.neo4j.graphdb.Node&nbsp;n)</PRE>
<DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getName()"><!-- --></A><H3>
getName</H3>
<PRE>
public java.lang.String <B>getName</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setName(java.lang.String)"><!-- --></A><H3>
setName</H3>
<PRE>
public void <B>setName</B>(java.lang.String&nbsp;value)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="hashCode()"><!-- --></A><H3>
hashCode</H3>
<PRE>
public int <B>hashCode</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>hashCode</CODE> in class <CODE>com.era7.bioinfo.bioinfoneo4j.BasicEntity</CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="equals(java.lang.Object)"><!-- --></A><H3>
equals</H3>
<PRE>
public boolean <B>equals</B>(java.lang.Object&nbsp;obj)</PRE>
<DL>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>equals</CODE> in class <CODE>com.era7.bioinfo.bioinfoneo4j.BasicEntity</CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="toString()"><!-- --></A><H3>
toString</H3>
<PRE>
public java.lang.String <B>toString</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>toString</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/FeatureTypeNode.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/EnzymeNode.html" title="class in com.era7.bioinfo.bio4j.model.nodes"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../../../com/era7/bioinfo/bio4j/model/nodes/GoTermNode.html" title="class in com.era7.bioinfo.bio4j.model.nodes"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../index.html?com/era7/bioinfo/bio4j/model/nodes/FeatureTypeNode.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="FeatureTypeNode.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>
Copyright &#169; 2012. All Rights Reserved.
</BODY>
</HTML>
