<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_19) on Wed Oct 24 11:41:10 CEST 2012 -->
<META http-equiv="Content-Type" content="text/html; charset=UTF-8">
<TITLE>
Uses of Package com.era7.bioinfo.bio4j.model.relationships.features (Bio4j 0.8 API)
</TITLE>

<META NAME="date" CONTENT="2012-10-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Package com.era7.bioinfo.bio4j.model.relationships.features (Bio4j 0.8 API)";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../../index.html?com/era7/bioinfo/bio4j/model/relationships/features/package-use.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="package-use.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H2>
<B>Uses of Package<br>com.era7.bioinfo.bio4j.model.relationships.features</B></H2>
</CENTER>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Packages that use <A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/package-summary.html">com.era7.bioinfo.bio4j.model.relationships.features</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.era7.bioinfo.bio4j.model.nodes"><B>com.era7.bioinfo.bio4j.model.nodes</B></A></TD>
<TD>General nodes&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.era7.bioinfo.bio4j.model.relationships.features"><B>com.era7.bioinfo.bio4j.model.relationships.features</B></A></TD>
<TD>Protein features relationships&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.era7.bioinfo.bio4j.programs"><B>com.era7.bioinfo.bio4j.programs</B></A></TD>
<TD>Bio4j importing process programs/modules&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.era7.bioinfo.bio4j.model.nodes"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/package-summary.html">com.era7.bioinfo.bio4j.model.relationships.features</A> used by <A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/nodes/package-summary.html">com.era7.bioinfo.bio4j.model.nodes</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/ActiveSiteFeatureRel.html#com.era7.bioinfo.bio4j.model.nodes"><B>ActiveSiteFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Amino acid(s) directly involved in the activity of an enzyme</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/SignalPeptideFeatureRel.html#com.era7.bioinfo.bio4j.model.nodes"><B>SignalPeptideFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sequence targeting proteins to the secretory pathway or periplasmic space</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/SpliceVariantFeatureRel.html#com.era7.bioinfo.bio4j.model.nodes"><B>SpliceVariantFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Amino acid change(s) producing alternate protein isoforms</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/TransmembraneRegionFeatureRel.html#com.era7.bioinfo.bio4j.model.nodes"><B>TransmembraneRegionFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Extent of a membrane-spanning region</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.era7.bioinfo.bio4j.model.relationships.features"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/package-summary.html">com.era7.bioinfo.bio4j.model.relationships.features</A> used by <A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/package-summary.html">com.era7.bioinfo.bio4j.model.relationships.features</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/BasicFeatureRel.html#com.era7.bioinfo.bio4j.model.relationships.features"><B>BasicFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Basic feature relationship</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.era7.bioinfo.bio4j.programs"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/package-summary.html">com.era7.bioinfo.bio4j.model.relationships.features</A> used by <A HREF="../../../../../../../com/era7/bioinfo/bio4j/programs/package-summary.html">com.era7.bioinfo.bio4j.programs</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/ActiveSiteFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>ActiveSiteFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Amino acid(s) directly involved in the activity of an enzyme</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/BindingSiteFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>BindingSiteFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Binding site for any chemical group (co-enzyme, prosthetic group, etc.)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/CalciumBindingRegionFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>CalciumBindingRegionFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Position(s) of calcium binding region(s) within the protein</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/ChainFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>ChainFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Extent of a polypeptide chain in the mature protein</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/CoiledCoilRegionFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>CoiledCoilRegionFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Positions of regions of coiled coil within the protein</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/CompositionallyBiasedRegionFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>CompositionallyBiasedRegionFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Region of compositional bias in the protein</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/CrossLinkFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>CrossLinkFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Residues participating in covalent linkage(s) between proteins</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/DisulfideBondFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>DisulfideBondFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Cysteine residues participating in disulfide bonds</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/DnaBindingRegionFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>DnaBindingRegionFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Position and type of a DNA-binding domain</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/DomainFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>DomainFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Position and type of each modular protein domain</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/GlycosylationSiteFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>GlycosylationSiteFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Covalently attached glycan group(s)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/HelixFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>HelixFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Helical regions within the experimentally determined protein structure</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/InitiatorMethionineFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>InitiatorMethionineFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Cleavage of the initiator methionine</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/IntramembraneRegionFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>IntramembraneRegionFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Extent of a region located in a membrane without crossing it</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/LipidMoietyBindingRegionFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>LipidMoietyBindingRegionFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Covalently attached lipid group(s)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/MetalIonBindingSiteFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>MetalIonBindingSiteFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Binding site for a metal ion</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/ModifiedResidueFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>ModifiedResidueFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Modified residues excluding lipids, glycans and protein cross-links</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/MutagenesisSiteFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>MutagenesisSiteFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Site which has been experimentally altered by mutagenesis</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/NonConsecutiveResiduesFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>NonConsecutiveResiduesFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Indicates that two residues in a sequence are not consecutive</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/NonStandardAminoAcidFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>NonStandardAminoAcidFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Occurence of non-standard amino acids (selenocysteine and pyrrolysine) in the protein sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/NonTerminalResidueFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>NonTerminalResidueFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The sequence is incomplete.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/NucleotidePhosphateBindingRegionFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>NucleotidePhosphateBindingRegionFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Nucleotide phosphate binding region</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/PeptideFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>PeptideFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Extent of an active peptide in the mature protein</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/PropeptideFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>PropeptideFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Part of a protein that is cleaved during maturation or activation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/RegionOfInterestFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>RegionOfInterestFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Region of interest in the sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/RepeatFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>RepeatFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Positions of repeated sequence motifs or repeated domains</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/SequenceConflictFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>SequenceConflictFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Description of sequence discrepancies of unknown origin</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/SequenceVariantFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>SequenceVariantFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Description of a natural variant of the protein</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/ShortSequenceMotifFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>ShortSequenceMotifFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Short (up to 20 amino acids) sequence motif of biological interest</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/SignalPeptideFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>SignalPeptideFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sequence targeting proteins to the secretory pathway or periplasmic space</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/SiteFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>SiteFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Any interesting single amino acid site on the sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/SpliceVariantFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>SpliceVariantFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Amino acid change(s) producing alternate protein isoforms</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/StrandFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>StrandFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Beta strand regions within the experimentally determined protein structure</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/TopologicalDomainFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>TopologicalDomainFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Location of non-membrane regions of membrane-spanning proteins</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/TransitPeptideFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>TransitPeptideFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Extent of a transit peptide for organelle targeting</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/TransmembraneRegionFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>TransmembraneRegionFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Extent of a membrane-spanning region</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/TurnFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>TurnFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Turns within the experimentally determined protein structure</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/UnsureResidueFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>UnsureResidueFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Regions of uncertainty in the sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../../../../../com/era7/bioinfo/bio4j/model/relationships/features/class-use/ZincFingerRegionFeatureRel.html#com.era7.bioinfo.bio4j.programs"><B>ZincFingerRegionFeatureRel</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Position(s) and type(s) of zinc fingers within the protein</TD>
</TR>
</TABLE>
&nbsp;
<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../../index.html?com/era7/bioinfo/bio4j/model/relationships/features/package-use.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="package-use.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>
Copyright &#169; 2012. All Rights Reserved.
</BODY>
</HTML>
