
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>bio4j</title>
  <meta name="author" content="oh no sequences!">

  
  <meta name="description" content="Hi everyone!A new version of Bio4j Go Tools viewer is available, it includes improvements in the graph visualization of GO annotation results.These...">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://bio4j.github.com/blog/page/3/">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="/javascripts/ender.js"></script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <link href="/atom.xml" rel="alternate" title="bio4j" type="application/atom+xml">
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href='http://fonts.googleapis.com/css?family=Open+Sans:600italic,300italic,400,400italic,600,300' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bree+Serif' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bitter:400,400italic' rel='stylesheet' type='text/css'>
<!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  showMathMenu: false,
  TeX: {
    extensions: ["AMSmath.js", "AMSsymbols.js"]
  }, 
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", webFont: "TeX", availableFonts: [] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>


  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-12161444-3']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   >
  <header role="banner"><!-- <hgroup>
  <h1><a href="/">bio4j</a></h1>
  
    <h2>bio data graph db</h2>
  
</hgroup>
 -->

 <hgroup>
  
    <a href="/">
      <img src="/images/logo.png">
    </a>
  <!-- <h2>bio data graph db</h2> -->

</hgroup>
</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
<form action="http://google.com/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:bio4j.github.com" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
<ul class="main-navigation">
  <li><a href="/blog">blog</a></li>
  <li><a href="/blog/archives">archives</a></li>
  <li><a href="/code">code</a></li>
  <li><a href="/docs">docs</a></li>
  <li><a href="/community">community</a></li>
  <li><a href="/about">about</a></li>
  </ul>

</nav>
  <div id="main">
    <div id="content">
      <div class="blog-index">
  
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/06/improvements-in-bio4j-go-tools-graph-visualization/">Improvements in Bio4j Go Tools (Graph visualization)</a></h1>
    
    
      <p class="meta">
        




  

<time datetime="2011-06-10T11:36:37-04:00" pubdate>Jun 10<span>th</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>Hi everyone!</p>

<p>A new version of <a href="http://gotools.bio4j.com:8080/Bio4jTestServer/Bio4jGoToolsWeb.html">Bio4j Go Tools viewer</a> is available, it includes improvements in the graph visualization of GO annotation results.
These are the new features:</p>

<ul>
  <li><strong>Load GO annotation results from URL</strong>: There’s no need anymore to upload the XML file with the results everytime you want to see the graph visualization. Just enter the publicly accessible URL of the file and the server will directly get the file for you.</li>
  <li>**Restrict **the visualization to only **one GO sub-ontology **at a time: Terms belonging to different sub-ontologies (<em>cellular component, biological process, molecular function</em>) are not mixed up anymore.</li>
  <li><strong>Choice of layout algorithms</strong>: You can choose between two different layout algorithms for the visualization, (<em>Yifan Hu</em> and <em>Fruchterman Reingold</em>).</li>
  <li><strong>Customizable layout algorithm time:</strong> Range of 1-10 minutes.</li>
</ul>

<p>I also made a short <strong><a href="http://www.bio4j.com/videos/Bio4jGoToolsGraphUseTutorial.htm">tutorial</a></strong> showing most of the features available in the following real-world use case: GO annotation results for <a href="https://github.com/ehec-outbreak-crowdsourced/BGI-data-analysis/wiki/Era7-annotation-of-bgi-v2-assembly-of-e.-coli-ty-2482">Era7 E. coli TY-2482 annotation with BG7 system of BGI V2 assembly</a></p>

<p>The corresponding GO Annotation results XML file is publicly available <strong><a href="https://s3-eu-west-1.amazonaws.com/pablo-tests/EHECAnnotationVersion2.xml">here</a></strong>. Just click the button ‘load file from url’ and paste the address of the file. </p>

<p>For those new to Bio4j Go Tools, two external open-source projects are used apart from Bio4j itself:</p>

<ul>
  <li><a href="http://gephi.org/toolkit/">Gephi Toolkit</a></li>
  <li><a href="https://github.com/jacomyal/SiGMa">SiGMa</a></li>
</ul>

<p>that’s all for now, keep an eye on the blog/twitter for updates ;)</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/05/bio4j-includes-refseq-data-now/">Bio4j includes RefSeq data now!</a></h1>
    
    
      <p class="meta">
        




  

<time datetime="2011-05-30T08:49:41-04:00" pubdate>May 30<span>th</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>Hi all,</p>

<p>After some weeks of hard work I finally finished the importer for RefSeq data.
First of all, I should clarify some points about its licensing:</p>

<ul>
  <li>Data has been retrieved from the public <a href="ftp://ftp.ncbi.nih.gov/refseq/release/complete/">ftp site</a> for RefSeq complete release. There is no extra/different data coming from other source.</li>
  <li>Quoting NCBI site: <em>“NCBI places no restrictions on the use or distribution of the GenBank data. However, some submitters may claim patent, copyright, or other intellectual property rights in all or a portion of the data they have submitted.”</em></li>
</ul>

<p>Once this has been said I will go into more details of how it’s been done.</p>

<h3 id="genome-elements-sequences">Genome elements’ sequences</h3>

<p>Sequences are not stored on Bio4j DB but uploaded as separate files to <a href="http://aws.amazon.com/s3/">S3</a> (Amazon Simple Storage Service) instead. Why doing it this way? For several reasons:</p>

<ul>
  <li>Having all sequences stored in the DB would take more than a decent amount of space</li>
  <li>Most queries to the DB wouldn’t be done in terms of the sequence content</li>
  <li>Relevant data included in RefSeq in terms of performing queries would be information about genes, rnas, genome elements, positions of all these elements, etc (rather than the sequence itself).</li>
  <li>Sequences are stored in txt files whose filename is the unique version string for the specific genome element, (e.g. NC_012932.txt) That way they can easily be retrieved whenever it’s needed. Plus, S3 service provides a way of extracting a range of bytes from a file without downloading the whole content, so there’s no need of downloading the complete sequence in the case where you already know the range of the sequence you are interested in.</li>
</ul>

<p>In some cases, no sequence can be uploaded for a genome element. These are the cases where instead of a final sequence, a list of terms as join(x…x)complement(x…x(contig(joing(…)) is provided (I never thought I’d find hundreds of lines with these terms where a sequence was supposed to be…).</p>

<h3 id="genome-elements-data">Genome elements’ data</h3>

<p>Regarding elements, the following are included (this are stored in Bio4j, not S3):</p>

<ul>
  <li>m RNA</li>
  <li>Misc RNA</li>
  <li>Nc RNA</li>
  <li>r RNA	</li>
  <li>t RNA</li>
  <li>rm RNA</li>
  <li>CDS</li>
  <li>gene</li>
</ul>

<p>Data stored for all these elements includes their positions and note attribute (whenever it’s found). I have to say that we decided not to extract more information from the gbff files since it can easily be accessed navigating through Bio4j by means of the connection Uniprot entry &lt;–&gt; RefSeq genome element. Plus, information included in Uniprot releases is much more reliable than that found in RefSeq files.</p>

</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/04/go-annotation-graph-visualizations-with-bio4j-go-tools-gephi-toolkit-sigma-project/">GO Annotation graph visualizations with Bio4j Go Tools + Gephi Toolkit + SiGMa project</a></h1>
    
    
      <p class="meta">
        




  

<time datetime="2011-04-25T12:30:01-04:00" pubdate>Apr 25<span>th</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>Hello everyone ;)</p>

<p>We’re back from Easter holidays bringing some cool graph visualization stuff. </p>

<p>Bio4j Go Tools includes now a new feature providing you with an interactive graph visualization for protein GO annotations.
The url of the app is still the <a href="http://gotools.bio4j.com:8080/Bio4jTestServer/Bio4jGoToolsWeb.html">same old one</a>.</p>

<p>On the server side, we’re using <a href="http://gephi.org/toolkit/">Gephi Toolkit</a> for applying layout algorithms while the corresponding Gexf file is generated with the class <a href="https://github.com/pablopareja/BioinfoUtil/blob/master/src/main/java/com/era7/lib/bioinfo/bioinfoutil/gephi/GephiExporter.java">GephiExporter</a> from <a href="https://github.com/pablopareja/BioinfoUtil">BioinfoUtil</a> project. The service is included in the project <a href="https://github.com/bio4j/Bio4jTestServer">Bio4jTestServer</a>, specifically the servlet <a href="https://github.com/bio4j/Bio4jTestServer/blob/master/src/java/com/era7/bioinfo/bio4j/server/servlet/GetGoAnnotationGexfServlet.java">GetGoAnnotationGexfServlet</a>.</p>

<p>Regarding to the client side, we’re using the open-source project <a href="https://github.com/jacomyal/SiGMa">SiGMa</a> for graph-visualization.</p>

<p>Here you have a screenshot of a small sample of GO Annotation results:</p>

<p><a href="http://gotools.bio4j.com:8080/Bio4jTestServer/Bio4jGoToolsWeb.html"><img src="/images/goAnnotationGraphScreenShot-1024x725.jpg" /></a></p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/04/bio4j-go-tools-goes-web/">Bio4j Go Tools goes web</a></h1>
    
    
      <p class="meta">
        




  

<time datetime="2011-04-11T12:43:50-04:00" pubdate>Apr 11<span>th</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>Hi!</p>

<p>From now on Bio4j Go Tools is available at the following <a href="http://gotools.bio4j.com:8080/Bio4jTestServer/Bio4jGoToolsWeb.html">address</a>.</p>

<p>This new version includes a chart for GO annotation terms frequency visualization. Some of the new chart features are:</p>

<ul>
  <li>GO terms search by name</li>
  <li>Independent ontology chart visualization</li>
  <li>Links to terms gene-ontology sites</li>
</ul>

<p><a href="http://gotools.bio4j.com:8080/Bio4jTestServer/Bio4jGoToolsWeb.html"><img src="/images/screenShotBlogGoTools-1024x762.jpg" /></a></p>

<p>In order to visualize your results just click on the “load file” button and select a XML file you previously generated using the GO Annotation service available in the first tab of the app.</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/03/bio4j-go-tools-first-version-available/">Bio4j Go Tools (first version available)</a></h1>
    
    
      <p class="meta">
        




  

<time datetime="2011-03-22T14:57:33-04:00" pubdate>Mar 22<span>nd</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>Hi everyone!</p>

<p>As you may have seen Bio4j has already started making his way through the bioinformatics world; however there’s not as much information as there should be about the project yet.
That’s where Bio4j Go Tools comes in as the first real-world example using Bio4j as back-end.</p>

<p>Bio4j Go Tools is a group of <a href="http://www.geneontology.org/">Gene Ontology</a> related services and apps. <em>(You can find more information about this in the <a href="http://wiki.bio4j.com/bio4jgotools">wiki</a>)</em></p>

<p>The services provided so far are:</p>

<ul>
  <li><a href="http://www.uniprot.org">Uniprot</a> protein GO annotations retrieving</li>
  <li>GoSlim requests with custom Slim term sets.</li>
</ul>

<p>Both services results and client-server communication are XML based following a really simple and intuitive structure.</p>

<p>A user-friendly <a href="https://github.com/pablopareja/Bio4jGoTools">AIR application</a> has been developed allowing the user to directly use these services abstracting the logic of the different requests.</p>

<p><a href="https://github.com/pablopareja/Bio4jGoTools"><img src="/images/Bio4jGoToolsScreenShot.jpg" width="739" title="Bio4j Go Tools AIR app screenshot" /></a></p>

<p>Enjoy it  ;)</p>

<h2 id="comments">comments</h2>

<ul>
  <li>
    <p><a href="http://alperyilmaz.org/blog"><strong>alper yilmaz</strong></a> 
When I put a protein id, the tools is asking for location of GoSlim.xml file, where can I retrieve this file? I checked the download page of Gene Ontology (http://www.geneontology.org/GO.downloads.ontology.shtml) but couldn’t find it.</p>

    <p>thanks.</p>

    <ul>
      <li><strong>Pablo Pareja</strong> Hi Alper,
You’re right it can be a bit confusing the way the app asks for a file location to save the results.
I just made some small changes to the app so that things are more straightforward. This version (v 1.01) is already available at Bio4jGoTools github repository</li>
    </ul>
  </li>
  <li>
    <p><a href="http://alperyilmaz.org/blog"><strong>alper yilmaz</strong></a> 
Nevermind, it was asking location and filename to save the results. So, there’s no problem..
Thanks..</p>
  </li>
</ul>

</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/02/new-neo4j-indexing-api-update/">New Neo4j Indexing API update</a></h1>
    
    
      <p class="meta">
        




  

<time datetime="2011-02-22T06:27:01-05:00" pubdate>Feb 22<span>nd</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>So the moment is approaching for Bio4j!</p>

<p>I’m updating the platform indexing system right now, putting it up to date with the new <a href="http://wiki.neo4j.org/content/Index_Framework">index API</a> for Neo4j. Once this change is made, there won’t be any other obstacles in the way for Bio4j to be launched…</p>

<p>Just wait a couple of days more…</p>
</div>
  
  


    </article>
  
  <div class="pagination">
    
    <a href="/blog/archives">Blog Archives</a>
    
    <a class="next" href="/blog/page/2/">Newer &rarr;</a>
    
  </div>
</div>
<aside class="sidebar">
  
    <section>
  <h1>Recent Posts</h1>
  <ul id="recent_posts">
    
      <li class="post">
        <a href="/blog/2014/02/bio4j-accepted-for-google-summer-of-code-2014/">Bio4j accepted for Google Summer of Code 2014</a>
      </li>
    
      <li class="post">
        <a href="/blog/2013/11/new-bio4j-success-berkeley-phylogenomics-grant/">Berkeley Phylogenomics Group receives an NSF grant to develop a graph DB for Big Data challenges in genomics building on Bio4j</a>
      </li>
    
      <li class="post">
        <a href="/blog/2013/10/bio4j-09-the-billion-relationships-is-here/">Bio4j 0.9 the billion relationships are here!</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/10/bio4j-modules-adapt-the-database-to-your-own-needs/">Bio4j modules, adapt the database to your own needs</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/10/bio4j-0-8-some-numbers/">Bio4j 0.8, some numbers</a>
      </li>
    
  </ul>
</section>

<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating&#8230;</li>
  </ul>
  
  <a href="https://github.com/bio4j">@bio4j</a> on GitHub
  
  <script type="text/javascript">
    $.domReady(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'bio4j',
            count: 0,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>



  
</aside>

    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2014 - oh no sequences! -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
</p>

</footer>
  


<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>




  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = 'http://platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
